src-java/stats-topology/stats-storm-topology/src/test/java/org/openkilda/wfm/topology/stats/StatsTopologyTest.java

```diff
@@ -845,6 +1426,183 @@ private void validateFlowStats(
+                    assertEquals(switchId.toOtsdFormat(), datapoint.getTags().get("switchid"));
+                    assertEquals(Integer.toString(flowStats.getOutPort()), datapoint.getTags().get("outPort"));
+                    assertEquals(Integer.toString(flowStats.getInPort()), datapoint.getTags().get("inPort"));
+                    break;
+                case METRIC_PREFIX + "haflow.ingress.packets":
+                case METRIC_PREFIX + "haflow.ingress.bytes":
+                case METRIC_PREFIX + "haflow.ingress.bits":
+                case METRIC_PREFIX + "haflow.ypoint.packets":
+                case METRIC_PREFIX + "haflow.ypoint.bytes":
+                case METRIC_PREFIX + "haflow.ypoint.bits":
+                case METRIC_PREFIX + "haflow.packets":
+                case METRIC_PREFIX + "haflow.bytes":
+                case METRIC_PREFIX + "haflow.bits":
+                    assertEquals(expectedEndpointTagsCount, datapoint.getTags().size());
+                    assertEquals(subFlowId, datapoint.getTags().get("flowid"));
+                    assertEquals(HA_FLOW_ID_1, datapoint.getTags().get("ha_flow_id"));
+                    assertEquals(direction, datapoint.getTags().get("direction"));
+                    break;
+                default:
+                    throw new AssertionError(format("Unknown metric: %s", datapoint.getMetric()));
+            }
+        });
+    }
+
+    private void validateHaFlowMeterStats(
+            MeterStatsEntry meterStats, FlowSegmentCookie cookie, SwitchId switchId, Long meterId, String subFlowId, boolean isYPoint) {
+        int expectedDatapointCount = 3;
+
+
+        List<Datapoint> datapoints = pollDatapoints(expectedDatapointCount);
+        //   List<Datapoint> datapoints = pollAllDataPoints();
+
+        Map<String, Datapoint> datapointMap = createDatapointMap(datapoints);
+
+
+        if (isYPoint) {
+            assertEquals(meterStats.getByteInCount() * 8,
+                    datapointMap.get(METRIC_PREFIX + "haflow.meter.ypoint.bits").getValue().longValue());
+            assertEquals(meterStats.getByteInCount(),
+                    datapointMap.get(METRIC_PREFIX + "haflow.meter.ypoint.bytes").getValue().longValue());
+            assertEquals(meterStats.getPacketsInCount(),
+                    datapointMap.get(METRIC_PREFIX + "haflow.meter.ypoint.packets").getValue().longValue());
+        } else {
+            assertEquals(meterStats.getByteInCount() * 8,
+                    datapointMap.get(METRIC_PREFIX + "haflow.meter.bits").getValue().longValue());
+            assertEquals(meterStats.getByteInCount(),
+                    datapointMap.get(METRIC_PREFIX + "haflow.meter.bytes").getValue().longValue());
+            assertEquals(meterStats.getPacketsInCount(),
+                    datapointMap.get(METRIC_PREFIX + "haflow.meter.packets").getValue().longValue());
+        }
```

Comment:
This `if` is incorrect, because meter can by `ingress` and `y point` in the sametime 
cases: `createHaFlow3` and `createHaFlow4`

Reply:
thanks fixed